toolbox/fdc3-for-web/fdc3-web-impl/src/handlers/IntentHandler.ts

```diff
@@ -116,16 +125,29 @@ class PendingIntent {
 
 export class IntentHandler implements MessageHandler {
   private readonly directory: Directory;
-  private readonly regs: ListenerRegistration[] = [];
+  private registrations: ListenerRegistration[] = [];
   readonly pendingIntents: Set<PendingIntent> = new Set();
-  readonly pendingResolutions: Map<string, AppIdentifier> = new Map();
+  readonly pendingResolutions: Map<string, FullAppIdentifier> = new Map();
   readonly timeoutMs: number;
 
   constructor(d: Directory, timeoutMs: number) {
     this.directory = d;
     this.timeoutMs = timeoutMs;
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  cleanup(instanceId: InstanceID, _sc: ServerContext<AppRegistration>): void {
+    this.registrations = this.registrations.filter(reg => reg.instanceId != instanceId);
+    //don't clean up pendingIntents as some apps may load
+
+    //cleanup pendingResolutions
+    this.pendingResolutions.forEach((val, key) => {
```

Comment:
this needs a test

Reply:
I don't see how we can test that meaningfully - other than by inspecting the state? Answers on a postcard.

Reply:
SHould be able to do it based on postedMessages again - I just noted that you currently send an error response if the pendingResolution doesn't exist - I'm changing that to return a success response and just not send on the result (as the app that raised the intent has gone away). That makes the test possible by checking that a `raiseIntentResultResponse` was NOT sent if the app had disconnected.

Reply:
Done that, which covers it.

Reply:
Added testing for this